packages/botonic-plugin-flow-builder/src/index.ts

```diff
@@ -165,8 +186,13 @@ export default class BotonicPluginFlowBuilder implements Plugin {
     }
     return result.target.id
   }
+
+  getPayloadParams<T>(payload: string): T & PayloadParamsBase {
+    const payloadParams = JSON.parse(payload.split(SEPARATOR)[1] || '{}')
+    return payloadParams
+  }
```

Comment:
Doubt: Is this method used anywhere? Why you can assure that the response will implement the `PayloadParamsBase` interface?

Reply:
I would like to discuss this with you. This method is for use in a bot action that runs when you receive a payload defined in a BotActionNode. I've been thinking about it and I think it needs to be typed like this

``typescript
  getPayloadParams<T extends PayloadParamsBase>(payload: string): T {
    const payloadParams = JSON.parse(payload.split(SEPARATOR)[1] || '{}')
    return payloadParams
  }
´´´

Reply:
applied